src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -1980,7 +1980,31 @@ private void LexSyntaxTrivia(bool afterFirstToken, bool isTrailing, ref SyntaxLi
                     case '#':
                         if (_allowPreprocessorDirectives)
                         {
-                            this.LexDirectiveAndExcludedTrivia(afterFirstToken, isTrailing || !onlyWhitespaceOnLine, ref triviaList);
+                            if (isTrailing || !onlyWhitespaceOnLine)
+                            {
+                                // Directives cannot be in trailing trivia.
+                                // We parse the directive (ignoring its effects like disabled text or defined symbols).
+                                // We extract the text corresponding to the parsed directive
+                                // and add it as a BadToken in SkippedTokensTrivia.
+
+                                var savePosition = TextWindow.Position;
+
+                                this.ParseDirective(isActive: false, endIsActive: false, isFollowingToken);
```

Comment:
>isActive: false, endIsActive: fals [](http://example.com/codeflow?start=52&length=34)

It would be good to explain why we use these values. What better effect do they have by comparison to other values.

Reply:
They don't have any effect - they are only written into the resulting directive syntax node but we are discarding it.

Reply:
However, since the directive is at invalid location, it seems better to parse it as "not active".

Reply:
Please add a comment reflecting the rational behind the choice